src/packages/solid/components/Keyboard/KeyboardInput.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSignal, type Component, type Signal } from 'solid-js';
+import { type IntrinsicNodeProps } from '@lightningjs/solid';
+// import styles, {
+//   type KeyboardInputStyleProperties,
+//   type KeyboardInputStyles
+// } from './KeyboardInput.styles.js';
+import Keyboard from './Keyboard.jsx';
+import Column from '../Column/Column.jsx';
+import Input from '../Input/Input.jsx';
+import type { KeyProps } from '../Key/Key.jsx';
+import type { Tone } from '../../types/types.js';
+import { type KeyboardStyles } from './Keyboard.styles.js';
+
+export type KeyboardFormat = Array<Array<string | KeyProps>>;
+
+export interface KeyboardInputProps extends IntrinsicNodeProps {
+  /**
+   * signal passed in to represent the actual title within the input
+   */
+  titleSignal: Signal<string>;
+
+  style?: Partial<KeyboardStyles>;
+
+  tone?: Tone;
+}
+
+// rows created from each array passed in
+const KeyboardInput: Component<KeyboardInputProps> = (props: KeyboardInputProps) => {
+  // signal representing a string of the key pressed and a boolean for if value has been added to the input yet
+  const keyEvent = createSignal('');
```

Comment:
Solid seems mad about this
```
For proper analysis, array destructuring should be used to capture the first result of this function call.eslint[solid/reactivity](https://github.com/solidjs-community/eslint-plugin-solid/blob/main/docs/reactivity.md)
```

is this actually fine or could it potentially cause a bug? @chiefcll